kaos/algorithm/interpolator.py

```diff
@@ -94,10 +96,20 @@ def interpolate(self, timestamp, kind="quadratic"):
         Raise:
             ValueError if interpolation could not be performed for the given timestamp.
         """
-        # find the correct segment
-        segment = OrbitSegment.get_by_platform_and_time(self.platform_id, timestamp)
+        # find the correct segment from stored data
+        segment = next((segment for segment in self.segments if segment.start_time <= timestamp
+                        and segment.end_time >= timestamp), None)
+
         if not segment:
-            raise InterpolationError("No segment found: {}, {}".format(self.platform_id, timestamp))
+            # segment was not in stored data, ask DB
+            segment = OrbitSegment.get_by_platform_and_time(self.platform_id, timestamp)
```

Comment:
Interesting approach to a pseudo sort of caching!

Reply:
66.957 seconds vs 37.058 seconds over 5-day intervals just with this change. Hopefully, it's not too interesting for SQLAlchemy.